### PR TITLE
Allow non-Observable hermitian generators

### PR DESCRIPTION
**Pull request by lillian542**
**Context:**
Operators should be able to to return generators that don't inherit from `Observable`. For example, `qml.s_prod(2.0, qml.PauliX(0))` should be a valid generator. Currently `qml.generator` checks if the generator found is an instance of Observable, and raises and error if not.

**Description of the Change:**
Replace check for whether the generator is `Observable` with a check that `generator.is_hermitian`. Modify the default return function (`_generator_prefactor`) to not insist 

**Benefits:**
Now calling `qml.generator` on `qml.ops.Evolution(qml.s_prod(0.5, qml.PauliX(0)), phase)` works.

**Possible Drawbacks:**
Not a drawback exactly, but qml.generator with format set to "hamiltonian" or "observable" will not work. Only the default "prefactor" is modified in this PR.

**Related GitHub Issues:**
#3473

### CODE REVIEW

File: tests/ops/op_math/test_exp.py [hunk 1]
```diff
@@ -799,3 +799,26 @@ def test_simplifying_Evolution_operator(self):
...
+    @pytest.mark.parametrize(
+        "base",
+        [
+            qml.pow(qml.PauliX(0) + qml.PauliY(1)),
+            qml.adjoint(qml.PauliZ(2)),
+            qml.s_prod(0.5, qml.PauliX(0)),
+        ],
+    )
+    def test_generator_not_observable_class(self, base):
+        """Test that the label is informative and uses decimals."""
```
Comment (rmoyard) [inline tests/ops/op_math/test_exp.py:812]: Comment to be changed I think.

File: tests/ops/op_math/test_exp.py [hunk 2]
```diff
@@ -799,3 +799,26 @@ def test_simplifying_Evolution_operator(self):
...
+        op = Evolution(base, 1)
+        gen = qml.generator(op)[0]
```
Comment (rmoyard) [inline tests/ops/op_math/test_exp.py:814]: Why do you need to index it here?

File: pennylane/ops/functions/generator.py [hunk 3]
```diff
@@ -78,11 +78,11 @@ def _generator_prefactor(gen, op):
...
     """
-    if isinstance(gen, (qml.Hermitian, qml.SparseHamiltonian)):
-        obs = gen
-        prefactor = 1.0
 
-    elif isinstance(gen, qml.operation.Observable):
+    obs = gen
+    prefactor = 1.0
+
+    if isinstance(gen, qml.operation.Observable):
```
Comment (rmoyard) [inline pennylane/ops/functions/generator.py:85]: I don't think you need the else, because you defined 'obs' and 'prefactor' before

**Review by rmoyard — COMMENTED** [review]
It looks good :100: I have some questions and suggestion, could you also specify what tests were changed for metric tensor? Also don't forget the changelog thx!

File: tests/transforms/test_metric_tensor.py [hunk 4]
```diff
@@ -1276,7 +1276,7 @@ def test_generator_no_expval(monkeypatch):
             qml.RX(np.array(0.5, requires_grad=True), wires=0)
             qml.expval(qml.PauliX(0))
 
-        with pytest.raises(qml.QuantumFunctionError, match="is not an observable"):
+        with pytest.raises(qml.QuantumFunctionError, match="is not hermitian"):
```
Comment (rmoyard) [inline tests/transforms/test_metric_tensor.py:1286]: Test looks good :+1:

File: tests/ops/op_math/test_exp.py (hunk 1 above)
Comment (lillian542) [reply tests/ops/op_math/test_exp.py:812]: Oops! Thanks :)

File: tests/returntypes/transforms/test_metric_tensor_new.py [hunk 5]
```diff
@@ -1276,7 +1276,7 @@ def test_generator_no_expval(monkeypatch):
             qml.RX(np.array(0.5, requires_grad=True), wires=0)
             qml.expval(qml.PauliX(0))
 
-        with pytest.raises(qml.QuantumFunctionError, match="is not an observable"):
+        with pytest.raises(qml.QuantumFunctionError, match="is not hermitian"):
```
Comment (rmoyard) [inline tests/returntypes/transforms/test_metric_tensor_new.py:1286]: Same :+1:

File: pennylane/ops/functions/generator.py (hunk 3 above)
Comment (lillian542) [reply pennylane/ops/functions/generator.py:85]: I believe the variable `gen` has been converted to a Hamiltonian inside the `if Observable` statement (in line 87), and so setting `obs=gen` inside the `else` statement is not equivalent to what was done previously.

**Comment by lillian542** [comment]
> could you also specify what tests were changed for metric tensor?

See the 3 commits titled 'Update tests' for specific changes to the test files :)

**Review by rmoyard — APPROVED** [review]
Thanks for answering my questions, it looks good to me :100:
